Develop - Linux/SQL Code Review

myanneng opened the pull request.

Commit:
Init project

myanneng committed:
Docker bash script.

myanneng committed:
Docker bash script.

myanneng committed:
Revert "Init project"

This reverts commit 79dd866e6cb341e6c7dab622b2b07cbde3674094.

myanneng committed:
Revert "Docker bash script."

This reverts commit 2edcd319f97b69292a879a435c1baf56e2da44cf.

"I accidentally used the wrong branch so I'm undoing this."

myanneng committed:
"Edited README.md file and uploaded architecture diagram."

myanneng committed:
Edited psql_docker.sh with additional commands.

myanneng committed:
Edited psql_docker.sh with additional commands. Adding it in the appropriate feature branch.

myanneng committed:
Implemented ddl.sql to create host_agent database, host_info, host_usage tables

myanneng committed:
Edited host_info.sh and host_usage.sh scripts to also insert their data into the psql database.

myanneng committed:
Added queries.

myanneng committed:
Edited host_usage.sh to add in password passing.

myanneng committed:
Added more information into the README.md beyond the architecture tasks.

myanneng committed:
Corrected markup mistakes.

myanneng committed:
Corrected the corrected markup mistakes.

myanneng committed:
Corrected the corrected markup mistakes.

myanneng committed:
Merge branch 'feature/architecture' into develop

myanneng committed:
Merge branch 'feature/postgreSQL' into develop

myanneng committed:
Merge branch 'feature/createDDL' into develop

myanneng committed:
Merge branch 'feature/monitoring_agent' into develop

myanneng committed:
Merge branch 'feature/queries' into develop

myanneng committed:
Fixed overarching README.md file for the portfolio to fix the link for the Linux/SQL project.

myanneng committed:
Created core_java project structure.

myanneng committed:
Fixed image URL.

myanneng committed:
Fixed command.

jarvisgroup commented on linux_sql/README.md:
```diff
@@ -0,0 +1,45 @@
+Cluster Monitor Agent is an internal tool that monitors the cluster resources. Irecords the hardware specifications of each node (server), and monitors its resource usage (e.g. CPU/memory) in realtime. The collected data is stored in a database found in a designated primary server node (Server no. 1). This tool allows for the Linux Cluster Administration (LCA) team to use the data to generate rep ...
+
+## Architecture and Design
+![Generalized architecture for the Cluster Monitor Agent](assets/architecture.png)
```
fix diagram

jarvisgroup commented on linux_sql/README.md:
```diff
@@ -0,0 +1,45 @@
+`host_usage.sh` is a script that is scheduled to run once a minute. It records the server's resource usage, specifically its CPU and memory utilization. This script also pushes this information to the database table `host_usage` for recording.
+
+## Usage 
+1. `psql_docker.sh`: there are two (2) input options for this script.
```
fix markdown syntax

jarvisgroup commented on linux_sql/scripts/host_info.sh:
```diff
@@ -0,0 +1,34 @@
+l2_cache=$(echo "$lscpu_output" | egrep "^L2 cache:" | awk '{print $3}' | xargs)
+total_mem=$(echo "$meminfo" | egrep "MemTotal:" | awk '{print $2}' | xargs)
+timestamp=$(vmstat -t | awk '{if(NR==3) print $18" "$19}' | xargs)
+
```
you need an env var for psql psswd

jarvisgroup commented on linux_sql/scripts/host_info.sh:
```diff
@@ -0,0 +1,34 @@
+database=$3
+psql_user=$4
+psql_password=$5
+
```
check # of args

jarvisgroup replied:
shell coding style

jarvisgroup commented on linux_sql/scripts/host_usage.sh:
```diff
@@ -0,0 +1,36 @@
+disk_io=$(vmstat --unit M | awk '{if(NR==3) print $9}' | xargs)
+disk_available=$(echo "$disk" | awk '{if(NR==3) print $4}' | xargs)
+
+export PGPASSWORD='$psql_password'; psql -h $hostname -U $psql_user -w $database_name -p $psql_port -c \
```
;
||
&&
|

jarvisgroup commented on linux_sql/scripts/psql_docker.sh:
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+# PSQL_Docker.sh script
+# The purpose of this script is to start/stop Docker and start/stop a PostgreSQL container. 
+
```
assign args to vars

jarvisgroup commented on linux_sql/scripts/psql_docker.sh:
```diff
@@ -0,0 +1,73 @@
+# If the command 'stop' is given, stop the running container.
+if [ "$1" = "stop" ];
+then
+   docker container stop jrvs-psql
```
check  daemon

jarvisgroup commented on linux_sql/scripts/psql_docker.sh:
```diff
@@ -0,0 +1,73 @@
+echo "Docker container started!"
+fi
+
+sudo yum install -y postgresql
```
remove 66-71

jarvisgroup commented on linux_sql/sql/queries.sql:
```diff
@@ -0,0 +1,45 @@
+GROUP BY
+	usage.host_id
+	info.hostname,
+	info.total_mem,
```
do you need group by total_mem

myanneng committed:
Fixed formatting.

myanneng committed:
Fixed sublist formatting.

myanneng committed:
Updated for formatting.

myanneng committed:
Initial design and implementation of grep app.

myanneng committed:
Revised and extended interface to use stream/lambda functions for two functions.

myanneng committed:
Edited a lambda function to be in more traditional format.

myanneng committed:
Fixed some of the IOExceptions and the  method to do something when an IOException is thrown.

myanneng committed:
Commented out something.

myanneng committed:
README.md file created and edited.

myanneng committed:
Fixed introduction header.

myanneng committed:
Re-uploading the header fix. It looks fine in the IDE, but not on the website.

myanneng committed:
Updated README.md

Fixed the header file directly. The problem was not the before/after line breaks, but lack of space character between "#" and "Introduction." My bad.

myanneng committed:
Updated README.md

Accidentally deleted a `.

myanneng committed:
Changed some formatting.

myanneng committed:
Edited README.md for Linux/SQL project.

myanneng committed:
Merge branch 'feature/readme' of https://github.com/jarviscanada/jarvis_data_eng_anne into feature/readme

Commit:
Merge branch 'feature/grep_lambda_implementation' into develop

Commit:
Merge branch 'feature/readme' into develop
